src/neuroconv/basedatainterface.py

```diff
@@ -60,13 +67,63 @@ def validate_metadata(self, metadata: dict) -> None:
         decoded_metadata = json.loads(serialized_metadata)
         validate(instance=decoded_metadata, schema=self.get_metadata_schema())
 
+    def get_default_backend_configuration(
+        self,
+        backend: Literal["hdf5", "zarr"] = "hdf5",
+        metadata: Optional[dict] = None,
+        **conversion_options,
+    ) -> Union[HDF5BackendConfiguration, ZarrBackendConfiguration]:
+        """
+        Fill and return a default backend configuration to serve as a starting point for further customization.
+
+        Parameters
+        ----------
+        backend : "hdf5" or "zarr", default: "hdf5"
+            The type of backend used to create the file.
+        metadata : dict, optional
+            Metadata dictionary with information used to create the NWBFile.
+        **conversion_options
+            Additional keyword arguments to pass to the `.add_to_nwbfile` method.
+        """
+        if metadata is None:
+            metadata = self.get_metadata()
+
+        with make_or_load_nwbfile(metadata=metadata, verbose=self.verbose) as nwbfile:
+            self.add_to_nwbfile(nwbfile=nwbfile, metadata=metadata, **conversion_options)
+            return get_default_backend_configuration(nwbfile=nwbfile, backend=backend)
+
     def create_nwbfile(self, metadata=None, **conversion_options) -> NWBFile:
+        """
+        The protocol for adding this interfaces data to a PyNWB neurodata type in the in-memory pynwb.NWBFile object.
+
+        Parameters
+        ----------
+        metadata : dict, optional
+            Metadata dictionary with information used to create the NWBFile.
+        **conversion_options
+            Additional keyword arguments to pass to the `.add_to_nwbfile` method.
+
+        Returns
+        -------
+        nwbfile : pynwb.NWBFile
+            The in-memory object with this interface's data added to it.
+        """
         nwbfile = make_nwbfile_from_metadata(metadata)
         self.add_to_nwbfile(nwbfile, metadata=metadata, **conversion_options)
         return nwbfile
 
     @abstractmethod
     def add_to_nwbfile(self, nwbfile: NWBFile, **conversion_options) -> None:
+        """
+        The protocol for adding this interfaces data to a PyNWB neurodata type in the in-memory pynwb.NWBFile object.
```

Comment:
```suggestion
        Add data from this interface to an in-memory pynwb.NWBFile object.
```

Reply:
The keyword protocol adds some context that 'this is where the PyNWB code goes' relevant to describing _how_ the data of this interface _maps_ onto the chosen neurodata object representations

Perhaps

"""
Define a protocol for mapping the data from this interface to NWB neurodata objects and add those objects to the in-memory pynwb.NWBFile?
"""

Reply:
@h-mayorquin Ideally, there would be a thorough tutorial about this included in the docs on constructing an interface - that would be nice all around and then here we could say "For more details and examples, see <cross link>".